File: dotnet/Barchart.BinarySerializer/Entities/Exceptions/EntityNotFoundException.cs
#region Using Statements

using Barchart.BinarySerializer.Entities.Keys;

#endregion

namespace Barchart.BinarySerializer.Entities.Exceptions;

/// <summary>
///     Thrown when an entity is not found in the entity manager.
/// </summary>
/// <typeparam name="TEntity">
///     The type of the entity.
/// </typeparam>
public class EntityNotFoundException<TEntity> : InvalidOperationException where TEntity : class, new()
{
    #region Constructor(s)

    /// <summary>
    ///     Initializes a new instance of the <see cref="EntityNotFoundException{TEntity}"/> class.
    /// </summary>
    /// <param name="key">
    ///     The key of the entity.
    /// </param>
    public EntityNotFoundException(IEntityKey<TEntity> key) : base($"The entity manager [ {nameof(TEntity)} ] does not contain the desired entity [ ${key} ].")
    {
        
    }
    
    #endregion
}

File: typescript/src/entities/exceptions/entity-not-found-exception.ts
import { EntityKeyDefinition } from "../keys/entity-key.interface";

/**
 * Thrown when an entity is not found in the entity manager.
 *
 * @public
 * @exported
 * @extends {Error}
 * @template TEntity - The type of the entity.
 */
export class EntityNotFoundException<TEntity extends object> extends Error {
    constructor(key: EntityKeyDefinition<TEntity>) {
        super(`The entity manager does not contain the desired entity [ ${key} ].`);

        Object.setPrototypeOf(this, EntityNotFoundException.prototype);
    }
}